Better visibility for players at a distance, and victory screen outline animation

Fueredoriku opened the pull request:
- Made players change LOD levels change at more reasonable ditstances
- Add inverse hull outline shader and material
- Made LOD3 players be transparent with inverse hull outline for better visibility
- Add jump fill outline shader and material
- Add jump fill outline to players seen from the arena camera (for spectating from far away + through geometry)
- Add jump fill outline animation to victory screen

Fueredoriku committed:
Fix playerLOD distance scale, add inverse hull outline

- Made players change LOD levels change at more reasonable ditstances
- Add inverse hull outline shader and material
- Made LOD3 players be transparent with inverse hull outline for better visibility

Fueredoriku committed:
Add jumpfill outline for players viewed from orbit camera

toberge commented on Assets/Shaders/RFOShader.shader:
perhaps some comment on top of this file with accreditation and license since this was from that article guy right?

Fueredoriku replied:
Added credits, license is a "do whatever you want", so I'll leave it at credits to article 👍

toberge commented on Assets/Shaders/RFO.cs:
also counts for parts of this file or?

Fueredoriku replied:
Yup

toberge commented on Assets/Scripts/Scoreboard/Scoreboard.cs:
```diff
@@ -229,4 +229,15 @@ private string FormatScore(int score)
+        gameObject.layer = layer;
+        foreach (Transform child in transform)
+        {
+            child.gameObject.layer = layer;
```
Not recursive?

Fueredoriku replied:
Don't actually need to because the outline shader pass creates a mask that already covers the relevant parts of the screen 🤠

toberge reviewed (approved):
Looks good for the most part, however

- Haven't inspected LOD stuff closely so only speaking about jump flood here, LOD seemed fine
- Victory outline animation has a jarring transition between thick black outline (very visible in some cases) and the extra thick white outline, which... kinda throw me off a little? Perhaps the swoosh to white should happen only when the camera has gone still (or just a little before), and perhaps we could avoid having the thick black outline displayed before it? (it cuts through the tower floor in crater town very noticeably)
- Feels like the player outlines on overview camera should be a different color than black? Perhaps white there as well? Looks kinda grimy :/ (my first thought here was to use the player colors but... this is already nice, that stuff can be attempted later)

Fueredoriku committed:
Add large outline animation for victory screen

Fueredoriku commented:
Changed up the victory animation a bit based on your feedback and it looks a lot better now imo. Also added white outlines to players on the overview camera
